### PR TITLE
More deployment refactoring

### PR DESCRIPTION
**Pull request by hadley**
* Remove unused code from `deploymentRecord()`, and return a list instead of data frame
* Perform all record value standardisation in `deploymentRecord()`
* Add more tests

To do:
* [ ] After review, organise functions in more logical order

### CODE REVIEW

File: R/deployments.R [hunk 1]
```diff
@@ -121,76 +113,55 @@ deployments <- function(appPath = ".",
...
+    appId = appId %||% "",
+    bundleId = bundleId %||% "",
+    url = url %||% "",
+    when = when,
+    lastSyncTime = lastSyncTime
+  )
+  c(standard, metadata)
 }
 
+# Workbench uses to show a list of recently deployed content on user dashboard
```
Comment (aronatkins) [inline R/deployments.R:150]: I had no idea that this was a thing!

File: R/deployments.R [hunk 2]
```diff
@@ -121,76 +113,55 @@ deployments <- function(appPath = ".",
...
-  write.dcf(deploymentRecord, newHistory, width = 4096)
-  cat("\n", file = newHistory, append = TRUE)
+  newHistory <- deploymentHistoryPath(new = TRUE)
+  writeDeploymentRecord(deploymentRecord, newHistory)
 
+  history <- deploymentHistoryPath()
   # append existing history to new history
-  if (file.exists(history))
+  if (file.exists(history)) {
+    cat("\n", file = newHistory, append = TRUE)
```
Comment (aronatkins) [inline R/deployments.R:162]: Why did the addition of a newline shift from its current location? Are we leaving a newly created DCF file without a trailing newline?

Expand the tests to confirm the behavior when creating the file and when appending to the file.

**Review by aronatkins — APPROVED** [review]

File: R/deployments.R (hunk 2 above)
Comment (hadley) [reply R/deployments.R:162]: This should be the newline that separates records in the dcf file; you're right that I should test the behaviour.

Comment (aronatkins) [reply R/deployments.R:162]: Thanks. I'd like to see the actual file contents, but it feels as if we have lost a trailing newline on newly created files.

My theory:

New: `second-record\nfirst-record`
Old: `second-record\nfirst-record\n`

Comment (hadley) [reply R/deployments.R:162]: I'm pretty sure `write.dcf()` adds that already, but I'll definitely double check.

**Pull request merged**